ambry-server/src/main/java/com/github/ambry/server/StatsManager.java

```diff
@@ -420,17 +377,13 @@ public void run() {
         if (!cancelled) {
           metrics.totalFetchAndAggregateTimeMs.update(time.milliseconds() - totalFetchAndAggregateStartTimeMs);
           // 4. Construct a StatsWrapper.
-          StatsHeader statsHeader = new StatsHeader(StatsHeader.StatsDescription.STORED_DATA_SIZE, time.milliseconds(),
-              partitionToReplicaMap.keySet().size(), partitionToReplicaMap.keySet().size() - unreachableStores.size(),
-              unreachableStores);
-          StatsWrapper statsWrapper = new StatsWrapper(statsHeader, aggregatedSnapshot);
+          StatsHeader statsHeader =
+              new StatsHeader(StatsHeader.StatsDescription.STORED_DATA_SIZE, time.milliseconds(), partitionIds.size(),
+                  partitionIds.size() - unreachableStores.size(), unreachableStores);
+          HostAccountStorageStatsWrapper statsWrapper =
+              new HostAccountStorageStatsWrapper(statsHeader, new HostAccountStorageStats(hostStorageStatsMap));
           // 5. Persist this statsWrapper to mysql database if connection exists.
-          if (accountStatsStore != null) {
-            accountStatsStore.storeAccountStats(statsWrapper);
-          }
-          // 6. Persist this statsWrapper to local backup file.
-          publish(statsWrapper);
-          logger.info("Local account stats snapshot published to {}", statsOutputFile.getAbsolutePath());
+          accountStatsStore.storeHostAccountStorageStats(statsWrapper);
```

Comment:
nit: I feel it's better to check accountStatsStore != null inside this method or constructor instead of outside of PartitionClassStatsPublisher.

Reply:
StatsManager would start two threads to run the publishers when the `accountStatStore` is not `null`. If we check the statement inside the publishers then we are creating the threads even if we don't do anything with the threads. That seems like a waste for threads.

Reply:
I see. thx for the explanation.